Let user know they put a invalid tenant id instead of saying unknown issue happen

jingcheng16 opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->

Example Sentry issue: https://dimagi.sentry.io/issues/5991125083/

From the stack trace, we can see it is because user put a wrong tenant id.

We didn't catch it, so we send a message saying we encountered an unknown issue, which is incorrect and misleading.
![image](https://github.com/user-attachments/assets/976fe7b4-847d-45c3-a6b1-384573cb6d11)


## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->

Ticket: https://dimagi.atlassian.net/browse/SAAS-16123

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Safe as we just add a try catch to catch errors.
I will also request QA on it.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

No

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
QA Ticket: https://dimagi.atlassian.net/browse/QA-7193
QA should change the configuration, to make the tenant id incorrect, and see if receive the "Auto deactivation skipped" email. The email context should explicitly say what's the issue, instead of saying unknown issue.


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

jingcheng16 committed:
Catch tenant id error

jingcheng16 committed:
Translate error message

nospame commented on corehq/apps/sso/utils/entra.py:
```diff
@@ -26,10 +26,17 @@ def get_all_usernames_of_the_idp_from_entra(idp):
+    except ValueError as e:
+        if "check your tenant name" in str(e):
+            raise EntraVerificationFailed(error="invalid_tenant",
+                                          message="Please double check your tenant id is correct")
```
Is there a reason to add our own message here rather than passing along the `ValueError` message that `msal` uses? Maybe this is more in line with other `EntraVerificationFailed` messages we raise.

nospame replied:
Nevermind, I see in the next commit that we do use our own messages for this.

jingcheng16 replied:
Thanks Evan! I find the long error message they returned is less helpful, based on our code, I can narrow down the reason is tenant id is wrong.

jingcheng16 committed:
pass the correct object

EntraVerificationFailed is an exception class, so it doesn’t have static attributes error or message......
